src/components/searchBar/index.tsx

```diff
@@ -1,23 +1,96 @@
 import React, { useCallback, useState } from "react";
 
+interface IngredientProps {
+  id: string;
+  name: string;
+}
+
 export default function SearchBar() {
-  const [value, setValue] = useState("");
+  const [searchQuery, setSearchQuery] = useState("");
+  const [ingredients, setIngredients] = useState<IngredientProps[]>([]);
+  const [selectedIngredients, setSelectedIngredients] = useState<IngredientProps[]>([]);
 
   const handleChangeInput = useCallback(
     (event: React.ChangeEvent<HTMLInputElement>) => {
       event.preventDefault();
-      setValue(event.target.value);
+      setSearchQuery(event.target.value);
+      fetch(
+        `https://api.spoonacular.com/food/ingredients/autocomplete?apiKey=${process.env.NEXT_PUBLIC_SPOONACULAR_API_KEY}&query=${event.target.value}&number=10`
+      )
+        .then((res) => res.json())
+        .then((result) => {
+          setIngredients(result);
+        });
     },
     []
   );
 
+  const generateKey = (pre: IngredientProps) =>  `${ pre.name.trim() }_${ new Date().valueOf() }`;
+
+
+  const handleSelectIngredient = useCallback(
+    (ingredient: IngredientProps) => {
+      if(selectedIngredients.some((selected) => selected.name === ingredient.name)){
+        alert('Ingredient already selected. Please, choose another one.')
+        return;
+      }
+      setSelectedIngredients((current) => [
+        ...current,
```

Comment:
```suggestion
      setSelectedIngredients((prev) => [
        ...prev,
```
This suggestion will make much more sense once you dig deep into refs.